src/arduino/arduino.ts

```diff
@@ -804,4 +839,118 @@ export class ArduinoApp {
             return false;
         });
     }
+
+    /**
+     * Private implementation. Not to be called directly. The wrapper burnBootloader()
+     * manages the busy state.
+     * @see https://arduino.github.io/arduino-cli/
+     * @see https://github.com/arduino/Arduino/issues/11765
+     * @remarks Currently this is only supported by `arduino-cli`. A request has been
+     * made with the Arduino repo.
+     */
+    private async _burnBootloader(): Promise<boolean> {
+        const dc = DeviceContext.getInstance();
+        const args: string[] = [];
+        let restoreSerialMonitor: boolean = false;
+        const verbose = VscodeSettings.getInstance().logLevel === constants.LogLevel.Verbose;
+
+        if (!this.boardManager.currentBoard) {
+            logger.notifyUserError("boardManager.currentBoard", new Error(constants.messages.NO_BOARD_SELECTED));
+            return false;
+        }
+        const boardDescriptor = this.boardManager.currentBoard.getBuildConfig();
+
+        if (this.useArduinoCli()) {
+            args.push("burn-bootloader",
+                "-b", boardDescriptor);
+        } else {
+            arduinoChannel.error("This command is only available when using the Arduino CLI");
+            return false;
+        }
+
+        if (!dc.port) {
+            await this._selectSerial();
+            return false;

```

Comment:
If I'm understanding this right, this forces someone to select the port, and then have to restart the burn bootloader process. Is this what you were going after?

Reply:
Yes, I was using the same flow found in `build()` for this just to be safe. I do like the idea of allowing burn bootloader or build operations continuing if a serial is selected though. It may be worth bringing this up in a separate issue.